okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelectorImpl.java

```diff
@@ -98,7 +98,8 @@ private static String switchWsToHttp(String url) {
 
     @Override
     public Optional<HttpUrl> redirectToNext(HttpUrl current) {
-        int index = currentUrl.updateAndGet(operand -> (operand + 1) % baseUrls.size());
+        int index = currentUrl.updateAndGet(
+                (currentUrlIndex) -> (indexFor(current).orElse(currentUrlIndex) + 1) % baseUrls.size());
```

Comment:
nit: running the `indexFor` inside the `compareAndSet` loop seems like it makes retries more likely, but maybe I'm prematurely optimising.

Reply:
yeah, pulled it out